channel.go

```diff
@@ -52,28 +49,26 @@ type Channel struct {
 	LastMessageAt time.Time `json:"last_message_at"`
 
 	client *Client
+	lock   *sync.Mutex
```

Comment:
No need for a pointer; just embed the struct.

Reply:
there is need for a pointer

Reply:
there is need pointer

Reply:
I don't see how pointers prevent deadlocks?

Reply:
if you use pointer - after `ch.update` mutex will be pointed to same object. So if other goroutine does waits for `Lock()` at the moment - it will be released after `Unlock()` happened. If we don't use pointer - after `ch.update` mutex will be another object and `Lock` will never released

Reply:
I still don't get it. The mutex applies/protects the Channel it lives in. They map 1 to 1. So when ch.update changes ch, then it uses the mutex embedded in ch, and only that one.

Reply:
https://github.com/GetStream/stream-chat-go/pull/8/files/a2340def2b75ef2f430ebfbade450d75b9ec944f..99b1adc4830130d92a2efc9db9586144bc3ecbac#diff-861e77702f49a71d32be4b3c7600818fR68

Reply:
https://github.com/GetStream/stream-chat-go/pull/8/files/a2340def2b75ef2f430ebfbade450d75b9ec944f..99b1adc4830130d92a2efc9db9586144bc3ecbac#diff-861e77702f49a71d32be4b3c7600818fR68

Reply:
what's your problem with pointer?

Reply:
It allows trickery. An embedded  struct enforces correct use. It documents the 1 to 1 relation.